### DISCUSSION TITLE
Unity 6

### DISCUSSION THREAD

**Discussion by XRWebland** [leapmotion/UnityModules#1647] [General]
Is there an ultra leap unity plugin for Unity 6? End set up we are looking at in the not to distant future is Unity 6 with MRTK 3 which is being upgraded for Unity 6 right now. Would love to test ultra leaps set up if needed.

**Comment by Miazyy** [comment]
已收到您的邮件，谢谢。祝好。周燕银

**Reply by MaxPalmer-UH** [reply]
Sort of. The issue with 6 is that the default render pipeline has switched to URP. Our samples and materials work with the built in render pipeline, not URP. There are a couple of samples scenes that use URP and some materials, but the examples and core assets are not URP aware. We've just released 7.2.0 which updates the scripts to support Unity 6 API changes though and you should be able to get things up and running on 6 with pretty low friction. We need to figure out how to best support Unity 6 / URP assets and Unity 2023 / Built in Render Pipeline assets. In the past it's not been easy to create materials that automatically support both, but you don't really want two copies of everything either.

**Reply by MaxPalmer-UH** [reply]
Also, MRTK3 is all in on OpenXR, so if you just want hand tracking data going through to MRTK3 then you don't necessarily require our plugin.

**Comment by MaxPalmer-UH** [comment]
An update on this, we're hoping to release 7.3 soon which has much smoother support for Unity 6.

**Reply by MaxPalmer-UH** [reply]
Sorry, this took waaay longer than expected. Release is now out.
